sdks/java/core/src/test/java/org/apache/beam/sdk/fn/data/BeamFnDataInboundObserverTest.java

```diff
@@ -146,10 +147,15 @@ public void testCloseVisibleToAwaitCompletionCallerAndProducer() throws Exceptio
             Arrays.asList(DataEndpoint.create(TRANSFORM_ID, CODER, (value) -> {})),
             Collections.emptyList());
 
+    AtomicBoolean isReady = new AtomicBoolean(false);
     Future<?> future =
         executor.submit(
             () -> {
               observer.accept(dataWith("ABC"));
+              synchronized (this) {
```

Comment:
If "this" object is used elsewhere for concurrent handing it could cause conflict. Could just synchronized (isReady) ?

Reply:
Good point. Changes are made.